tests/unit/test_current_user.py

```diff
@@ -0,0 +1,94 @@
+import pytest
+from bokeh.document import Document
+from dataclasses import dataclass
+
+from mz_bokeh_package.utilities import CurrentUser
+from mz_bokeh_package.utilities import MZGraphQLClient
+
+API_KEY = "6fzQxEJL1YX9ldWUhHv5NPpiIcV7Gg8r03KwmeCqyRakj2nSAstDMTb4oB"
+SESSION_ID = "LME0EBfddYQCJqpZoO4ysYvynDXFLavC8hu3RzHW6e5e"
+USER_ID = "79e8e0f4-d2d6-4c16-b833-0a8986a6ce53"
+USER_NAME = "user_name"
+
+
+TESTS = [
+    # Test 1: session ID provided and is logged in CurrentUser._users. API key not requered.
+    {
+        "input": {
+            "session_active": True,
+            "logged_user": True,
+            "session_id": SESSION_ID,
+            "api_key": None,
+        },
+        "output": {"id": USER_ID, "name": USER_NAME}
+    },
+    # Test 2: session ID provided and is not logged in CurrentUser._users. API key provided.
+    {
+        "input": {
+            "session_active": True,
+            "logged_user": False,
+            "session_id": SESSION_ID,
+            "api_key": API_KEY,
+        },
+        "output": {"id": USER_ID, "name": USER_NAME}
+    },
+    # Test 3: session ID not provided and is not logged in CurrentUser._users. API key provided.
+    {
+        "input": {
+            "session_active": False,
+            "logged_user": False,
+            "session_id": None,
+            "api_key": API_KEY,
+        },
+        "output": {"id": USER_ID, "name": USER_NAME}
+    },
+    # Test 4: session ID not provided and is not logged in CurrentUser._users. API key not provided.
+    {
+        "input": {
+            "session_active": False,
+            "logged_user": False,
+            "session_id": None,
+            "api_key": None,
+        },
+        "output": ValueError
+    },
+]
+
+
+@dataclass
+class SessionContext:
+    id: str = ""
+
+
+@pytest.mark.parametrize("parameters", TESTS)
+def test_get_user_info(monkeypatch, parameters: dict):
+    CurrentUser._users = {}
+    session_active = parameters['input']['session_active']
```

Comment:
After you implement all the other changes I suggested in this module, I think you don't need to set these variables anymore. I think each input parameter will only be needed once, and you can just use `parameters['input']['session_active']` directly instead of `session_active`. Your test should simplify to just a few lines of code.